portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,18 +35,70 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<>();
-    comments.add("Hey !!! Just commenting - Prakash");
-    comments.add("Roses are red - Nirmala");
-    comments.add("Violets are blue - Sushant");
-    comments.add("What I am doing, I have no clue- Prashant");  
-    
+
+    Query query = new Query("Comment");
+    List<Entity> comments = new ArrayList<>();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    Iterator<Entity> commentIterator = results.asIterator();
+    int limit = 0;
+
+    int maxComment;
+    try {
+      maxComment = Integer.parseInt(request.getParameter("max-comment"));
+    } catch (NumberFormatException e) {
+      maxComment = Integer.MAX_VALUE;
+    }
```

Comment:
Extract this to another method. `getCommentLimit` or something.

Reply:
Changes made.